### PR TITLE
Reference doc generation

### PR DESCRIPTION
**Pull request by timotheeguerin**
Recreating this PR so we can get the website automatically published and view result 

Examples:
-  [Rest](https://cadlwebsite.z1.web.core.windows.net/prs/1415/standard-library/rest/reference.html)
-  [OpenAPI](https://cadlwebsite.z1.web.core.windows.net/prs/1415/standard-library/openapi/reference.html)

### CODE REVIEW

**Review by allenjzhang — APPROVED** [review]

File: packages/ref-doc/src/experimental.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { compile, joinPaths, NodeHost } from "@cadl-lang/compiler";
+import { mkdir, writeFile } from "fs/promises";
+import { renderToDocusaurusMarkdown } from "./emitters/docusaurus.js";
+import { extractRefDocs } from "./extractor.js";
+
+/**
+ * @experimental this is for experimental and is for internal use only. Breaking change to this API can happen at anytime.
```
Comment (nguerrera) [inline packages/ref-doc/src/experimental.ts:7]: Good call! 

Curious, does IDE do anything interesting with this or is just that you'll see it in hover/signature-help/intellisense with the docs?

Comment (timotheeguerin) [reply packages/ref-doc/src/experimental.ts:7]: I don’t think it does. I was thinking of using deprecated to have it show crossed but that also felt a little wrong.

the package is not published anyway so don’t think it’s a big deal

File: packages/ref-doc/src/utils/type-signature.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  const value = typeof member.value === "string" ? `"${member.value}"` : member.value;
+  return value === undefined ? `${ns}${member.name}` : `${ns}${member.name}: ${value}`;
+}
+
+export function getQualifier(parent: Type | undefined) {
+  if (!parent) {
+    return "";
+  }
+
+  const parentName = getTypeName(parent);
```
Comment (nguerrera) [inline packages/ref-doc/src/utils/type-signature.ts:137]: I was just noticing in the server copy of this where operation (not decorator or function) signatures show `(anonymous model).paramName`. I was going to fix it. Not for right now, but there is probably some opportunity to consolidate signature printing with some callbacks or options, not sure.

It looks like maybe we don't show interface operations yet in the ref docs? I tried to see if this was impacting this too but in the REST docs for example, I see the interfaces but not the operations in them.

Comment (timotheeguerin) [reply packages/ref-doc/src/utils/type-signature.ts:137]: yeah for now I didn't show it, in the same way as I don't show model properties, enum members, etc.. I think that can be a follow up as might need a bit of dicussion on how we want to show it

**Review by nguerrera — APPROVED** [review]
This is awesome!

**Pull request merged**